packages/jest-resolve-dependencies/src/index.ts

```diff
@@ -66,23 +62,27 @@ class DependencyResolver {
   }
 
   resolveInverseModuleMap(
-    paths: Set<Path>,
-    filter: (file: Path) => boolean,
-    options?: ResolveModuleConfig,
-  ): Array<ResolvedModule> {
+    paths: Set<Config.Path>,
+    filter: (file: Config.Path) => boolean,
+    options?: Resolve.ResolveModuleConfig,
+  ): Array<Resolve.ResolvedModule> {
     if (!paths.size) {
       return [];
     }
 
-    const collectModules = (related, moduleMap, changed) => {
+    const collectModules = (
+      related: Set<Config.Path>,
+      moduleMap: Array<Resolve.ResolvedModule>,
+      changed: Set<Config.Path>,
+    ) => {
       const visitedModules = new Set();
-      const result: Array<ResolvedModule> = [];
+      const result: Array<Resolve.ResolvedModule> = [];
       while (changed.size) {
         changed = new Set(
-          moduleMap.reduce((acc, module) => {
+          moduleMap.reduce<Array<Config.Path>>((acc, module) => {
             if (
               visitedModules.has(module.file) ||
-              !module.dependencies.some(dep => dep && changed.has(dep))
+              !module.dependencies.some(dep => changed.has(dep))
```

Comment:
`dep &&` can return `''`, but `some` wants `boolean`.
I've just removed the check, the typings don't allow nullish values and `''` is not a relevant case.